ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -0,0 +1,301 @@
+import * as React from 'react';
+
+import {Component} from '@angular/core';
+
+import {Button, Clickable} from 'app/components/buttons';
+import {FlexColumn, FlexRow} from 'app/components/flex';
+import {ClrIcon} from 'app/components/icons';
+import {TextInput} from 'app/components/inputs';
+import {workspaceAdminApi} from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import {ReactWrapperBase} from 'app/utils';
+import {
+  getSelectedPopulations,
+  getSelectedResearchPurposeItems
+} from 'app/utils/research-purpose';
+
+import {
+  AdminWorkspaceResources,
+  UserRole,
+  Workspace
+} from 'generated/fetch';
+
+
+const styles = {
+  fixedWidthWithMargin: {
+    width: '10rem',
+    marginRight: '1rem'
+  },
+  dropdownIcon: {
+    marginRight: '1rem',
+    transform: 'rotate(180deg)',
+    transition: 'transform 0.5s',
+    color: colors.primary
+  },
+  dropdownIconOpen: {
+    transform: 'rotate(0deg)',
+  }
+};
+
+const FlexColumnWithRightMargin = ({style = {}, children}) => {
+  return <FlexColumn style={{...styles.fixedWidthWithMargin, ...style}}>
+    {...children}
+  </FlexColumn>;
+};
+
+const LabelWithPrimaryColor = ({style = {}, children}) => {
+  return <label style={{color: colors.primary, ...style}}>
+    {...children}
+  </label>;
+};
+
+interface State {
+  googleProject: string;
+  workspace: Workspace;
+  collaborators: Array<UserRole>;
+  resources: AdminWorkspaceResources;
+  detailsOpen: boolean;
+}
+
+export class AdminWorkspace extends React.Component<{}, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      googleProject: '',
+      workspace: null,
+      collaborators: [],
+      resources: {
```

Comment:
How important is it to create a deep-but-empty object here? IMO it might save a bit of maintenance over time to leave this null to start and not have to worry about deep-constructing the object. (But I guess that's a trade-off against the cost of doing qualified indexing in the render section... totally fine if you ended up concluding that was more important.)